modules/node-shared/src/main/scala/io/constellationnetwork/node/shared/domain/delegatedStake/UpdateDelegatedStakeAcceptanceManager.scala

```diff
@@ -0,0 +1,84 @@
+package io.constellationnetwork.node.shared.domain.delegatedStake
+
+import cats.data.NonEmptyChain
+import cats.data.Validated.{Invalid, Valid}
+import cats.effect.Async
+import cats.syntax.all._
+
+import io.constellationnetwork.node.shared.domain.delegatedStake.UpdateDelegatedStakeValidator.{
+  UpdateDelegatedStakeValidationError,
+  UpdateDelegatedStakeValidationErrorOr
+}
+import io.constellationnetwork.schema.delegatedStake.UpdateDelegatedStake
+import io.constellationnetwork.schema.epoch.EpochProgress
+import io.constellationnetwork.schema.{GlobalSnapshotInfo, SnapshotOrdinal}
+import io.constellationnetwork.security.SecurityProvider
+import io.constellationnetwork.security.signature.Signed
+import io.constellationnetwork.syntax.sortedCollection.sortedMapSyntax
+
+import eu.timepit.refined.types.numeric.NonNegLong
+
+trait UpdateDelegatedStakeAcceptanceManager[F[_]] {
+
+  def accept(
+    creates: List[Signed[UpdateDelegatedStake.Create]],
+    withdrawals: List[Signed[UpdateDelegatedStake.Withdraw]],
+    lastSnapshotContext: GlobalSnapshotInfo,
+    lastGlobalEpochProgress: EpochProgress,
+    lastSnapshotOrdinal: SnapshotOrdinal
+  ): F[UpdateDelegatedStakeAcceptanceResult]
+
+}
+
+object UpdateDelegatedStakeAcceptanceManager {
+  def make[F[_]: Async: SecurityProvider](validator: UpdateDelegatedStakeValidator[F]) =
+    new UpdateDelegatedStakeAcceptanceManager[F] {
+      def accept(
+        creates: List[Signed[UpdateDelegatedStake.Create]],
+        withdrawals: List[Signed[UpdateDelegatedStake.Withdraw]],
+        lastSnapshotContext: GlobalSnapshotInfo,
+        lastGlobalEpochProgress: EpochProgress,
+        lastSnapshotOrdinal: SnapshotOrdinal
+      ): F[UpdateDelegatedStakeAcceptanceResult] = {
+
+        def partitionAccepted[A](validated: List[UpdateDelegatedStakeValidationErrorOr[A]], signed: List[A]) =
+          validated
+            .zip(signed)
```

Comment:
`Validated` monad may contain the the type that is being verified. here is that `[A]`. I think zipping like this is error prone, as you would have to ensure that both lists are equally the same length and both are ordered. instead, you can the the value from validated monad. so, having the list of validated elements, you should partition them by valid/invalid at that's it